### PR TITLE
Angular -> React: MySidePanel 

### PR DESCRIPTION
**Pull request by seanfeld**
# Angular -> React: MySidePanel

## Description
Added `MySidePanel` component and integrated it with angular controllers styling etc. Still Work In progress but making this to get eyes on it. 

## Findings
This code is very hard to deal with and debug. (I'll add more to this list when I'm less tired)
- The css naming is all under nested classes and parents so it takes a while to debug once you finish writing the react code.
- The import systems we use today (webpack) are nonexistent in the angular code so everything is one giant angular glob + things are put onto the window to be used elsewhere. - Our build system is very confusing
- Can't pass angular controller into react components as props.
- The window (global) has way too much stuff added to it and is probably extremely inefficient. 
- Had to copy certain functions and add things to the controllers/css styling in order to have 1:1.

Things that still need to be fixed: https://cdap.atlassian.net/browse/CDAP-18830 trakced here
- [ ] Ordering of plugins
- [ ] Code Cleanup - break sidePanel into small components

## PR Type
- [ ] Bug Fix
- [X] Feature
- [ ] Build Fix
- [ ] Testing
- [X] General Improvement

## Current

Screen Shot 2022-02-22 at 12 38 17 AM

Screen Shot 2022-02-22 at 12 38 31 AM

Previous iteration:
Screen Shot 2022-01-19 at 10 54 05 PM

### CODE REVIEW

File: app/cdap/components/hydrator/components/MySidePanel/MySidePanel.tsx [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+import { shouldShowCustomIcon, getCustomIconSrc, filterPlugins, generateLabel } from './helpers';
+
+interface IMySidePanelProps {
+  itemGenericName: string;
+  groups: any;
+  groupGenericName: string;
+  onPanelItemClick: any;
+}
+
+export const MySidePanel = ({
```
Comment (njbriggs) [inline app/cdap/components/hydrator/components/MySidePanel/MySidePanel.tsx]: Nit: drop `My` from names

File: app/cdap/components/hydrator/components/SidePanel/SidePanel.tsx [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+  const handleClickPlugin = (event, plugin) => {
+    // the angular controller function has event as an input
+    onPanelItemClick(event, plugin);
+  };
+
+  const renderIconPlugins = (plugins: [any]) => {
+    return plugins.map((plugin) => {
+      const showCustomIcon = shouldShowCustomIcon(plugin, pluginsMap);
+      return (
+        <div
```
Comment (njbriggs) [inline app/cdap/components/hydrator/components/SidePanel/SidePanel.tsx:141]: Use Mui `IconButton`?

File: app/cdap/components/hydrator/components/SidePanel/SidePanel.tsx [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+        <span className="plugin-badge">T</span>
+      </div>;
+    });
+  };
+
+  const renderGroups = () => {
+    // ordering still not the same
+    return groups.map((group, i) => {
+      const thisGroupOpened = openedGroup === group.name;
+      return (
```
Comment (njbriggs) [inline app/cdap/components/hydrator/components/SidePanel/SidePanel.tsx:193]: Could we adapt Mui `Accordion` for groups?

File: app/cdap/components/hydrator/components/SidePanel/SidePanel.tsx [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+  Typography,
+} from '@material-ui/core';
+import AppsIcon from '@material-ui/icons/Apps';
+import ListIcon from '@material-ui/icons/List';
+import styled, { css } from 'styled-components';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+
+interface ISidePanelProps {
+  itemGenericName: string;
+  groups: any;
```
Comment (mrahanjam) [inline app/cdap/components/hydrator/components/SidePanel/SidePanel.tsx]: nit: can we avoid use of `any` here and on line 42 if possible?

File: app/cdap/components/hydrator/components/SidePanel/SidePanel.tsx [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+        <StyledAccordion elevation={0} square defaultExpanded={i === 0}>
+          <StyledAccordionSummary
+            expandIcon={<ExpandMoreIcon />}
+            aria-controls=""
+            id={`${group.name}-acc-summary`}
+            data-cy={`plugin-${group.name}-group`}
+          >
+            <Chip label={group.plugins.length} size="small" />
+            <Typography
+              style={{
```
Comment (mrahanjam) [inline app/cdap/components/hydrator/components/SidePanel/SidePanel.tsx]: Should we avoid inline styling for the sake of consistency (on line 237 too)?

**Review by mrahanjam — APPROVED** [review]
Left a couple of unblocking comments, LGTM!

File: app/cdap/components/hydrator/components/SidePanel/SidePanel.tsx [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+    return plugins.map((plugin) => {
+      const showCustomIcon = shouldShowCustomIcon(plugin, pluginsMap);
+      return (
+        <div
+          role="button"
+          className={`plugin-item ${plugin.nodeClass ? plugin.nodeClass : ''} ${
+            plugin.hovering ? 'hovered' : ''
+          }`}
+          // need to implement my popover
+          // my-popover
```
Comment (njbriggs) [inline app/cdap/components/hydrator/components/SidePanel/SidePanel.tsx:147]: Did you get the popover working for this? Is this comment still needed?

Comment (seanfeld) [reply app/cdap/components/hydrator/components/SidePanel/SidePanel.tsx:147]: @njbriggs popover is being tracked in another ticket

**Pull request merged**